### DISCUSSION TITLE
Any equivalent to KDE's Activities for i3?

### DISCUSSION THREAD

**Discussion by knauth** [i3/i3#5976] [Q&A]
I recently switched from Plasma + Bismuth to i3. In Plasma, my workflow relied heavily on Activities- I would use one for work, one for social stuff, etc, with 4 workspaces within each. This allowed me to avoid "workspace bloat" where there are 6+ workspaces open for a variety of things, making it easy to lose track of what's where. It's a useful top-level abstraction to help me organize my work spatially. Is there any way to duplicate this functionality in i3? The only thing I can think of is setting shortcuts to go to the 5th and 9th workspaces and calling those the first workspace in a new activity, which seems hacky.

**Comment by KebabLord** [comment]
i use this approach to collect irc,mail, telegram, steamchat etc. on, let's say "social" workspace:
```ini
# Social workspace
bindsym $mod+o workspace Social
assign [class="hexchat"] workspace Social
assign [class="Mail"] workspace Social
assign [class="xelegram"] workspace Social
assign [title="Friends List"] workspace Social
bindsym $mod+Shift+o move container to workspace Social
```

So in this way:
- When i press Mod+o it will focus on my "social" activity / workspace
- When one of this applications started, they will be automatically created on this "social" workspace because of the assignment

I think this is literally KDE's Activity but you are configuring it yourself and assigning applications by their "xprop" values.

For more about configuration:
https://i3wm.org/docs/userguide.html#assign_workspace
https://unix.stackexchange.com/a/362061

**Comment by waterkip** [comment]
I'm actually in the process of building such a thing for myself. [I call them "Workspace groups"](https://github.com/i3/i3/discussions/5646), but I think we are somewhat talking about the same thing. In my case, I have a couple of workspaces for work, client, and personal use. They all have similar layouts.

I currently have a very small POC where I apply layouts to workspaces, and it looks at the layout to determine which applications are automatically started.

My idea is to "hide" the inactive workspaces and make the active group workspaces available on the 1, 2, 3, and 4 keys while the rest aren't. With tick events, I want to be able to switch groups. It requires some additional configuration.

**Reply by waterkip** [reply]
Here is a first draft of my tooling: 

https://metacpan.org/dist/AnyEvent-I3X-Workspace-OnDemand

The README in the git repo is a bit more clear on how to configure it/use it:
https://gitlab.com/waterkip/anyevent-i3x-workspace-ondemand/-/blob/master/README.md

My own script is found here, it is started once i3 starts
https://gitlab.com/waterkip/bundle-waterkip/-/blob/i3/bin/i3-workspace-thing?ref_type=heads
